test/util/TestFeatureRegistry.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <memory>
+
+#include "../../source/config/Config.h"
+#include "../../source/devicedefender/DeviceDefenderFeature.h"
+#include "../../source/util/FeatureRegistry.h"
+#include "gtest/gtest.h"
+
+using namespace std;
+using namespace Aws::Iot::DeviceClient;
+using namespace Aws::Iot::DeviceClient::Util;
+using namespace Aws::Iot::DeviceClient::DeviceDefender;
+
+class TestFeatureRegistry : public ::testing::Test
+{
+  public:
+    shared_ptr<FeatureRegistry> features;
+    shared_ptr<DeviceDefenderFeature> deviceDefender;
```

Comment:
You can invent dummy features for this test. Use something like to below to create `FakeFeature` instances with different names so that you can test the api with more than 1 entry in the registry. Overall you need more tests.

```
class FakeFeature : public Feature
{
  public:
    FakeFeature(const std::string& name) :name(name) {}
    void start() override { }
    void stop() override { }
    std::string getName() { return name; }
  private:
     std::string name;
};
```